src/runtime/components/forms/Form.vue

```diff
@@ -207,95 +203,162 @@ function isSuperStructSchema(schema: any): schema is Struct<any, any> {
       path: issue.path.join('.'),
       message: issue.message
     }))
+
+    return {
+      errors,
+      result: null
+    }
+  }
+  return {
+    result: result.data,
+    errors: null
   }
-  return []
 }
 
-function isJoiSchema(schema: any): schema is JoiSchema {
-  return schema.validateAsync !== undefined && schema.id !== undefined
-}
+async function validateSuperstructSchema(state: any, schema: Struct<any, any>): Promise<ValidateReturnSchema<typeof state>> {
+  const [err, result] = schema.validate(state)
+  if (err) {
+    const errors = err.failures().map(error => ({
+      message: error.message,
+      path: error.path.join('.')
+    }))
 
-function isJoiError(error: any): error is JoiError {
-  return error.isJoi === true
+    return {
+      errors,
+      result: null
+    }
+  }
+
+  return {
+    errors: null,
+    result
+  }
 }
 
-async function getJoiErrors(
+async function validateYupSchema(
   state: any,
-  schema: JoiSchema
-): Promise<FormError[]> {
+  schema: YupObjectSchema<any>
+): Promise<ValidateReturnSchema<typeof state>> {
   try {
-    await schema.validateAsync(state, { abortEarly: false })
-    return []
+    const result = schema.validateSync(state, { abortEarly: false })
+    return {
+      errors: null,
+      result
+    }
   } catch (error) {
-    if (isJoiError(error)) {
-      return error.details.map(detail => ({
-        path: detail.path.join('.'),
-        message: detail.message
+    if (isYupError(error)) {
+      const errors = error.inner.map(issue => ({
+        path: issue.path ?? '',
+        message: issue.message
       }))
+
+      return {
+        errors,
+        result: null
+      }
     } else {
       throw error
     }
   }
 }
 
-function isValibotSchema(schema: any): schema is ValibotSchema30 | ValibotSchemaAsync30 | ValibotSchema31 | ValibotSchemaAsync31 | ValibotSafeParser31<any, any> | ValibotSafeParserAsync31<any, any> | ValibotSchema | ValibotSchemaAsync | ValibotSafeParser<any, any> | ValibotSafeParserAsync<any, any> {
-  return '_parse' in schema || '_run' in schema || (typeof schema === 'function' && 'schema' in schema)
-}
-
-async function getValibotError(
-  state: any,
-  schema: ValibotSchema30 | ValibotSchemaAsync30 | ValibotSchema31 | ValibotSchemaAsync31 | ValibotSafeParser31<any, any> | ValibotSafeParserAsync31<any, any> | ValibotSchema | ValibotSchemaAsync | ValibotSafeParser<any, any> | ValibotSafeParserAsync<any, any>
-): Promise<FormError[]> {
-  const result = await ('_parse' in schema ? schema._parse(state) : '_run' in schema ? schema._run({ typed: false, value: state }, {}) : schema(state))
-  return result.issues?.map(issue => ({
-    // We know that the key for a form schema is always a string or a number
-    path: issue.path?.map(item => item.key).join('.') || '',
-    message: issue.message
-  })) || []
+function parseSchema(state: any) {
+  return async (schema: Schema): Promise<ValidateReturnSchema<typeof state>> => {
```

Comment:
I find this function a bit confusing, do you really need to return a function? Can't we call it directly with `schema` and `props.state` to get the list of errors directly?

Here's an example:
```ts
function getSchemaErrors(schema: Schema, state: any): Promise<ValidateReturnSchema<typeof state>> {
    if (isZodSchema(schema)) {
      return validateZodSchema(state, schema)
    } else if (isJoiSchema(schema)) {
      return validateJoiSchema(state, schema)
    } else if (isValibotSchema(schema)) {
      return validateValibotSchema(state, schema)
    } else if (isYupSchema(schema)) {
      return validateYupSchema(state, schema)
    } else if (isSuperStructSchema(schema)) {
      return validateSuperstructSchema(state, schema)
    } else {
      throw new Error('Form validation failed: Unsupported form schema')
    }
}
```